fuse/utils/multiprocessing/linux_shared_memory.py

```diff
@@ -0,0 +1,92 @@
+import os
+from os.path import join, basename
+import shutil
+
+# from multiprocessing import Lock
+from filelock import FileLock
+from glob import glob
+import psutil
+
+SHARED_MEM_FILES_PREFIX = "OUR_SHARED_MEM_@"
+SHARED_MEM_ACTIVE_ENV_VAR = "ACTIVATE_OUR_SHARED_MEM"
+SHM_BASE_DIR = "/dev/shm/"
+
+G_lock = FileLock(join(SHM_BASE_DIR, "our_shared_mem_file_lock"))  # Lock()
```

Comment:
Why do we need a global lock? shouldn't per copied file lock work?

Reply:
I can do it per file, but where should it be initialized ?
also - if we don't keep it sequential like now, the free RAM check won't be correct anymore